azure-common/src/main/java/com/azure/common/configuration/ClientConfiguration.java

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.common.configuration;
+
+import com.azure.common.credentials.AsyncServiceClientCredentials;
+import com.azure.common.http.HttpClient;
+import com.azure.common.http.policy.HttpLogDetailLevel;
+import com.azure.common.http.policy.HttpPipelinePolicy;
+import com.azure.common.http.policy.RetryPolicy;
+import org.slf4j.ILoggerFactory;
+import org.slf4j.LoggerFactory;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Client configuration settings common to Azure {@link com.azure.common.ServiceClient}. Settings such as
+ * {@link HttpClient}, retry policy, service endpoint, etc. can be set here.
+ */
+public class ClientConfiguration {
+    private HttpClient client;
+    private AsyncServiceClientCredentials credentials;
+    private String userAgent;
+    private RetryPolicy retryPolicy;
+    private List<HttpPipelinePolicy> policies;
+    private HttpLogDetailLevel httpLogDetailLevel;
+    private ILoggerFactory loggerFactory;
+    private URL serviceEndpoint;
+
+    /**
+     * Gets the default configuration settings
+     */
+    public ClientConfiguration() {
+        this.retryPolicy = new RetryPolicy();
+        this.policies = new ArrayList<>();
+        this.httpLogDetailLevel = HttpLogDetailLevel.NONE;
+        this.loggerFactory = LoggerFactory.getILoggerFactory();
+    }
+
+    /**
+     * Returns the credentials used to authenticate HTTP requests sent.
+     *
+     * @return Gets the credentials that will be used by ServiceClient to authenticate with.
+     */
+    public AsyncServiceClientCredentials credentials() {
+        return credentials;
+    }
+
+    /**
+     * Sets the credentials to use when authenticating HTTP requests.
+     *
+     * @param credentials The credentials to use for authenticating HTTP requests.
+     * @return The updated ClientConfiguration object.
+     */
+    public ClientConfiguration credentials(AsyncServiceClientCredentials credentials) {
+        Objects.requireNonNull(credentials);
```

Comment:
Whenever you have an unchecked exception, please add an `@throws` JavaDoc statement that explains what triggers that exception to arise, e.g. `@throws NullPointerException if {@code credentials} is {@code null}`